planner/core/exhaust_physical_plans.go

```diff
@@ -1886,8 +1886,22 @@ func (p *LogicalJoin) tryToGetMppHashJoin(prop *property.PhysicalProperty, useBC
 		lPartitionKeys, rPartitionKeys := p.GetPotentialPartitionKeys()
 		if prop.MPPPartitionTp == property.HashType {
 			var matches []int
-			if matches = prop.IsSubsetOf(lPartitionKeys); len(matches) == 0 {
+			if p.JoinType == InnerJoin {
```

Comment:
do previous tests not cover these cases?

Reply:
I guess so.